src/target/ch32f1.c

```diff
@@ -139,11 +139,14 @@ static bool ch32f1_flash_unlock(target *t)
 	uint32_t cr = target_mem_read32(t, FLASH_CR);
 	if (cr & FLASH_CR_FLOCK_CH32) {
 		DEBUG_WARN("Fast unlock failed, cr: 0x%08" PRIx32 "\n", cr);
-		return true;
+		return false ;
 	}
-	return false;
+	return true;
 }
-
```

Comment:
Please keep the blank line between the two functions. Additionally we aren't using Doxygen (and even if we were, it'd be with `@brief`, etc), please use a normal block comment and append *'s on the front of continuation lines as in:
```c
/*
 * Attempt to fast lock CH32F1 devices
 */
 ```